src/main/java/org/dita/dost/module/XsltModule.java

```diff
@@ -215,4 +214,26 @@ public void setExtension(final String extension) {
         this.extension = extension.startsWith(".") ? extension : ("." + extension);
     }
 
+    private void configureExtensions (TransformerFactory tf) {
+        if (tf.getClass().isAssignableFrom(net.sf.saxon.TransformerFactoryImpl.class)) {
+            configureSaxonExtensions((net.sf.saxon.TransformerFactoryImpl) tf);
+        }
+    }
+
+    // Registering full integrated function definitions (instances of net.sf.saxon.lib.ExtensionFunctionDefinition abstract class).
+    // https://www.saxonica.com/html/documentation/extensibility/integratedfunctions/ext-full-J.html
+    private void configureSaxonExtensions(net.sf.saxon.TransformerFactoryImpl tfi) {
+        net.sf.saxon.Configuration conf = tfi.getConfiguration();
```

Comment:
Use `final` variable declarations

Reply:
Could also maybe import `net.sf.saxon.TransformerFactoryImpl`?

Reply:
I intentionally use fully qualified name to avoid the intersection with Xalan's TransformerFactoryImpl. XSLT module supports both processors.